src/arch/armv8/inc/arch/gic.h

```diff
@@ -438,7 +438,7 @@ size_t gich_num_lrs();
 static inline size_t gic_num_irqs()
 {
     size_t itlinenumber = bit32_extract(gicd->TYPER, GICD_TYPER_ITLN_OFF, GICD_TYPER_ITLN_LEN);
-    return 32 * itlinenumber + 1;
+    return 32 * itlinenumber - 1;
```

Comment:
Although the original operation is indeed wrong, the correct formula is 32 * (N + 1).
Which should be:
```suggestion
    return 32 * (itlinenumber - 1);
```

Reply:
Yes, I agree with this. Check page 89 of GICv2 specification.

Reply:
Note that this function returns the number of interrupts, and not the range.